api/src/commonMain/kotlin/dev/kord/cache/api/observables/DataCache.kt

```diff
@@ -0,0 +1,43 @@
+package dev.kord.cache.api.observables
+
+/**
+ * An implementation of [TypedCache] that stores a set of [EntryCache]s.
+ */
+public class DataCache : TypedCache {
+
+    /**
+     * The set of [EntryCache]s stored in this [TypedSetCache].
+     */
+    private val types: MutableSet<EntryCache<*>> = mutableSetOf()
+
+    /**
+     * Returns the [EntryCache] for the specified type, or `null` if it is not found.
+     */
+    private suspend fun <T : Any> getTypeOrNull(): EntryCache<T>? {
+        return toSet().firstNotNullOfOrNull { it as? EntryCache<T> }
```

Comment:
this cast will still always succeed because of type erasure

Reply:
I'm aware I pushed that file by mistake while fixing the other issues.
I'm trying to find a way that doesn't use reflection in this context

Reply:
maybe we should use some explicit key for a `EntryCache` instead? could e.g. be an interface and use a companion object:
```
interface TypedCacheKey<T>

data class MemberData(...) {
    companion object : TypedCacheKey<T>
}

// get a cache:
typedCache.get(MemberData)
// or even a operator fun:
typedCache[MemberData]
```

Reply:
or do we even need `TypedCache` at all? we could replace `Kord.cache` with a new type inside of kord itself that is just something like:
```kotlin
class KordCache {
    val users: EntryCache<UserData>
    // ... repeat for all other cached types
}
```

Reply:
You are right it sounds redundant I was essentially thinking of a central access point

Reply:
Done.